docusaurus/docs/recipes/video-player.md

```diff
@@ -0,0 +1,99 @@
+---
+id: video-player
+title: Adding a video player
+sidebar_label: Video player
+---
+
+It's often normal to have the necessity of having videos in your website. In this recipe we will show you how to use [`react-player`](https://github.com/CookPete/react-player) and some common use cases.
+
+## Walk-through
+
+### 1. Install `react-player`
+
+```sh
+npm i react-player
+```
+
+### 2. Use `ReactPlayer` in your project
+
+You just need to import `ReactPlayer` and start using it.
+
+```js
+// ...
+import ReactPlayer from 'react-player';
+
+const Home = () => (
+    <div className={ styles.home }>
+        <ReactPlayer url="https://www.video-url.com" />
+    </div>
+);
+```
+
+### 3. Usage
+
+Now let's go through some common use cases that might be helpful.
+
+> ⚠️ By default `react-player` sets a width of 640px and a height of 360px to the player. You should change it in order to fit your needs.
+
+#### 1. Autoplay
+
+To enable autoplay you need to use `playing` prop.
+
+```js
+<ReactPlayer url="https://www.video-url.com" playing />
+```
+
+Be sure to check the different browser policies for autoplaying videos as they are different and you might need additional configuration. For example on Chrome you might need to mute the video in order to autoplay it.
```

Comment:
Mentioning Chrome is ok, but we should definitely mention Safari as well, which is a fair bit more inconsistent and cumbersome with video, especially in low power mode mobile devices.

Reply:
Sure. Can you give more info on that? I am not familiar with those problems.

Reply:
Sure, in Webkit when a device is in low power mode autoplay seems to be inconsistently disabled. You can find [examples here](https://stackoverflow.com/questions/50400902/detect-if-ios11-device-is-in-low-power-mode-to-prevent-bad-ux-on-normally-correc) of people struggling with autoplay in low power mode. This is compounded by there being no good way to verify if a device is in low power mode or if the video stopped playing. In an earlier project i hacked up a solution that pools the video's ready state until enough data to play continuously, or give up after an interval, with a condition like so:

```js
videoRef.current.getInternalPlayer().readyState >= HTMLMediaElement.HAVE_CURRENT_DATA
```

I don't think these are worth mentioning since they're fairly obscure and hacky, but they're just an example of struggling with Webkit because of obscure or undocumented behaviors.